plugins/chat/assets/javascripts/discourse/components/chat-composer.js

```diff
@@ -395,7 +395,20 @@ export default Component.extend(TextareaTextManipulation, {
         treatAsTextarea: true,
         autoSelectFirstSuggestion: true,
         transformComplete: (v) => v.username || v.name,
-        dataSource: (term) => userSearch({ term, includeGroups: true }),
+        dataSource: (term) => {
+          return userSearch({ term, includeGroups: true }).then((result) => {
+            if (result?.users?.length > 0) {
+              const presentUserNames =
```

Comment:
This could be a Set.

Reply:
Yes, totally should be set @martin-brennan tsk. Maybe also the next commit?